packages/lexical-code/src/CodeHighlighter.ts

```diff
@@ -468,43 +469,65 @@ function isEqual(nodeA: LexicalNode, nodeB: LexicalNode): boolean {
   );
 }
 
+/**
+ * Returns a boolean.
+ * Check that the selection span is within a single CodeNode.
+ * This is used to guard against executing handlers that can only be
+ * applied in a single CodeNode context
+ */
 function $isSelectionInCode(selection: null | BaseSelection): boolean {
   if (!$isRangeSelection(selection)) {
     return false;
   }
   const anchorNode = selection.anchor.getNode();
+  const maybeAnchorCodeNode = $isCodeNode(anchorNode)
+    ? anchorNode
+    : anchorNode.getParent();
   const focusNode = selection.focus.getNode();
-  if (anchorNode.is(focusNode) && $isCodeNode(anchorNode)) {
-    return true;
-  }
-  const anchorParent = anchorNode.getParent();
-  return $isCodeNode(anchorParent) && anchorParent.is(focusNode.getParent());
+  const maybeFocusCodeNode = $isCodeNode(focusNode)
+    ? focusNode
+    : focusNode.getParent();
+
+  return (
+    maybeAnchorCodeNode.is(maybeFocusCodeNode) &&
+    $isCodeNode(maybeAnchorCodeNode)
```

Comment:
```suggestion
    $isCodeNode(maybeAnchorCodeNode) &&
    maybeAnchorCodeNode.is(maybeFocusCodeNode)
```

The type guard should go before calling methods that require it to be true